opds_import.py

```diff
@@ -260,53 +321,135 @@ def import_from_metadata(
             edition, self.metadata_client, replace=policy
         )
 
+        return edition
+
+
+    def import_pools_works_from_circulationdata(
+            self, circulationdata, even_if_no_author, cutoff_date, immediately_presentation_ready
+    ):
+        """ For the passed-in CirculationData object, see if can find or create a license pool, 
+        then see if can find one or more editions pre-created for that license pool, 
+        then see if can create a work, and also update the editions 
+        with the new license pool and work.  
+        """
+
+        # Locate or create a LicensePool for this book.
+        license_pool, is_new_license_pool = circulationdata.license_pool(self._db)
+        work = None
+
+        if (not (cutoff_date 
+            and not is_new_license_pool 
+            and license_pool.last_checked < cutoff_date)):
```

Comment:
I think this is wrong. The right thing to check here is `metadata.last_update_time`. We're trying to avoid doing work if the OPDS entry hasn't changed since the last time we checked. In this case `cutoff_date` stands in for "the last time we checked". You could argue that `license_pool.last_checked` is a better proxy for 'the last time we checked', but right now you're comparing two values that both basically mean 'the last time we checked'. You want to compare 'the last time we checked' to 'the last time the OPDS entry changed server-side'.

This means adding a last_update_time to CirculationData that is the same value you give to Metadata. It might also make sense to not use `cutoff_date` here, and instead compare `license_pool.last_checked` to `last_update_time`.